src/PointwiseFunctions/GeneralRelativity/CMakeLists.txt

```diff
@@ -71,8 +71,12 @@ spectre_target_headers(
 
 target_link_libraries(
   ${LIBRARY}
-  PUBLIC DataStructures
-  INTERFACE ErrorHandling
+  PUBLIC
+  DataStructures
+  Domain
```

Comment:
I only find 2 includes of "Domain/Tags.hpp" and 1 include of "Domain/FunctionsOfTIme/Tags.hpp" in GeneralizedHarmonic/. The former 2 are in SecondTimeDerivOfSpacetimeMetric.*pp and can just be deleted I think. The latter is in ConstraintGammas.hpp and should be moved to Systems/GeneralizedHarmonic/ConstraintDaming/DampingFunction.hpp I think. Then the dependence on `Domain` is eliminated here.

Reply:
Here's what I find:
```
File: Psi4Real.hpp
  11 #include "Domain/Tags.hpp"

File: GeneralizedHarmonic/SecondTimeDerivOfSpacetimeMetric.hpp
  12 #include "Domain/Tags.hpp"

File: GeneralizedHarmonic/SecondTimeDerivOfSpacetimeMetric.cpp
   9 #include "Domain/Tags.hpp"

File: GeneralizedHarmonic/ConstraintGammas.hpp
  13 #include "Domain/FunctionsOfTime/Tags.hpp"
```

I agree, the `SecondTimeDerivOfSpacetimeMetric` can be deleted so I'll do that. The rest I'll let someone else clean up. The `Psi4Real` is a hard dependency on domain because of tags.


We have a few issues open to clean up the GH code and nobody's done it yet. Moving things to the system doesn't help because this directory circularly depends on the system one:

```
in: "~/spectre/src/PointwiseFunctions/GeneralRelativity/"
File: GeneralizedHarmonic/ConstraintGammas.hpp
  14  #include "Evolution/Systems/GeneralizedHarmonic/ConstraintDamping/Tags.hpp"
  15  #include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"

File: GeneralizedHarmonic/GaugeSource.hpp
  11  #include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"

File: GeneralizedHarmonic/TimeDerivOfSpatialMetric.hpp
  11  #include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"

File: GeneralizedHarmonic/TimeDerivOfLapse.hpp
  12  #include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"

File: GeneralizedHarmonic/TimeDerivOfShift.hpp
  12  #include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"

File: GeneralizedHarmonic/Pi.hpp
  12  #include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"

File: GeneralizedHarmonic/ExtrinsicCurvature.hpp
  13  #include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"

File: GeneralizedHarmonic/SecondTimeDerivOfSpacetimeMetric.hpp
  12  #include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"

File: GeneralizedHarmonic/DerivSpatialMetric.hpp
  12  #include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"

File: GeneralizedHarmonic/Phi.hpp
  12  #include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"

File: GeneralizedHarmonic/CovariantDerivOfExtrinsicCurvature.hpp
  10  #include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"

File: GeneralizedHarmonic/SpatialDerivOfLapse.hpp
  12  #include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"

File: GeneralizedHarmonic/SpatialDerivOfShift.hpp
  11  #include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"

File: Surfaces/ComputeItems.hpp
  16  #include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
```

Fixing all that is a lot more work and I think beyond the scope of this PR.